public/app/features/migrate-to-cloud/onprem/Page.tsx

```diff
@@ -1,43 +1,172 @@
+  };
+}
+
 export const Page = () => {
-  const { data: status, isFetching } = useGetStatusQueryMock();
-  const { data: resources } = useListResourcesQueryMock(status?.enabled ? undefined : skipToken);
-  const [startMigration, { isLoading: startMigrationIsLoading, isError: startMigrationIsError }] =
-    useStartMigrationMutationMock();
-  const [isDisconnecting, setIsDisconnecting] = useState(false);
+  const migrationDestination = useGetLatestMigrationDestination();
+  const migrationRun = useGetLatestMigrationRun(migrationDestination.data?.id);
+  const [actuallyRunMigration, actuallyRunMigrationResult] = useRunCloudMigrationMutation();
+
+  const isBusy = actuallyRunMigrationResult.isLoading || migrationDestination.isFetching || migrationRun.isFetching;
+  // const resources: MigrationResourceDTOMock[] = [];
+
+  // TODO: API returns this in a *very* wrong format - got to unmarshall it
+  const resources = useMemo(() => {
+    if (!migrationRun.data) {
+      return undefined;
+    }
 
-  if (!status?.enabled) {
+    const rawResources: { items: Array<MigrateDataResponseItemDto & { type: string }> } = JSON.parse(
+      /// @ts-expect-error
+      atob(migrationRun.data.result)
+    );
+
+    // converts API status to our expected/mocked status
+    function convertStatus(status: string) {
+      switch (status) {
+        case 'OK':
+          return 'migrated';
+        case 'failed':
+          return 'failed';
+        default:
+          return 'failed';
+      }
+    }
+
+    const betterResources: MigrationResourceDTOMock[] = rawResources.items.flatMap((item) => {
+      if (item.type === 'DATASOURCE') {
+        const datasourceConfig = Object.values(config.datasources).find((v) => v.uid === item.refId);
+
+        return {
+          uid: item.refId ?? '',
+          status: convertStatus(item.status ?? ''),
+          type: 'datasource',
+          resource: {
+            uid: item.refId ?? '',
+            name: datasourceConfig?.name ?? 'Unknown data source',
+            type: datasourceConfig?.meta?.name ?? 'Unknown type',
+            icon: datasourceConfig?.meta?.info?.logos?.small,
+          },
+        };
+      }
+
+      if (item.type === 'DASHBOARD') {
+        return {
+          uid: item.refId ?? '',
+          status: convertStatus(item.status ?? ''),
+          type: 'dashboard',
+          resource: {
+            uid: item.refId ?? '',
+            name: item.refId ?? 'Unknown dashboard',
+          },
+        };
+      }
+
+      return [];
+    });
+
+    return betterResources;
+  }, [migrationRun.data]);
+
+  console.log('migration run resources', resources);
+
+  const handleDisconnect = useCallback(() => {
+    window.alert('TODO: Disconnect');
+  }, []);
+
+  const handleStartMigration = useCallback(() => {
+    if (migrationDestination.data?.id) {
+      actuallyRunMigration({ id: migrationDestination.data?.id });
+    } else {
+      window.alert('id still aint there');
```

Comment:
how is it possible to reach this branch? 🤔

Reply:
it shouldn't be, strictly. At the moment the codegen makes types with all properties optional (something to look into)